### ISSUE TITLE
Rename no-warning-comments to no-comments and make it fixable

### ISSUE THREAD

**Issue by villesau** [eslint/eslint#10976]
**What rule do you want to change?**
[no-warning-comments](https://eslint.org/docs/rules/no-warning-comments)

**Does this change cause the rule to produce more or fewer warnings?**
Same amount.

**How will the change be implemented? (New option, new default behaviour, etc.)?**
By renaming rule to `no-comments`, and possibly making it fixable (removing the comment) and to support custom messages.

**Please provide some example code that this change will affect:**
https://github.com/flow-typed/flow-typed/pull/2834/files
Here you can see comment starting `// flow-typed signature` which is auto generated when fetching the libdef. This is something that should not be committed to repository, but often happens still. `no-warning-comments` rule is very good fit to prevent this, but the comment in question is not actually warning: It's just autogenerated hash of the libdef to make it possible to identify wether it's changed or not. 

```js
// flow-typed signature: ba506076507fa389f4f828349e657080
```

**What does the rule currently do for this code?**
`1:1  error  Unexpected 'flow-typed signature' comment  no-warning-comments`
**What will the rule do after it's changed?**
`1:1  error  Unexpected 'flow-typed signature' comment  no-comments`
or
`1:1  error  Unexpected 'flow-typed signature' <custom-message-like: this comment is auto generated and should not be committed to repository>`

Warnings are not only kind of comments that might be disallowed. By removing the `warning` from the rule name, it would make the rule more generic, and semantically correct in more use cases. To further improve the rule, it could support `--fix` which would remove the comment, and it could support custom messages. In our case it could be something like `flow-typed signature is autogenerated and should not be committed to repository`

**Comment by villesau** [comment]
On the other hand I wouldn't say that the examples like preventing `TODO: `s etc are warnings either, they are some sort of reminders instead.

I think some rules allows to define custom message. If that would be allowed for this rule, developer could decide what to communicate about the restriction: `1:1 error Unexpected 'TODO: ' No TODO comments, please finalize your work before committing to repository` etc.

**Comment by kaicataldo** [comment]
It sounds like you're advocating for something like `no-restricted-comments` that would be configured with a mapping of a string/RegExp to an an error message (similar to [no-restricted-syntax](https://eslint.org/docs/rules/no-restricted-syntax) but for comments). That seems like a reasonable change to me, since the new rule would cover everything the current rule does but allow for greater flexibility (and that we have a very similar rule for syntax already).
